File: client/src/types/JsonModels/AdminAI/AICostSummaryModel.ts
import type { JsonModels } from "../../backend";

export interface AICostSummaryModel
{
	from: string;
	to: string;
	estimatedCost: number;
	inputTokens: number;
	outputTokens: number;
	cachedInputTokens: number;
	byDay: JsonModels.AdminAI.AICostByDayModel[];
	byModel: JsonModels.AdminAI.AICostByModelModel[];
	byPlan: JsonModels.AdminAI.AICostByPlanModel[];
}

File: server/FitMate.Core/JsonModels/AdminAI/AICostSummaryModel.cs
namespace FitMate.Core.JsonModels.AdminAI;

public class AICostSummaryModel
{
    public DateTime From { get; set; }
    public DateTime To { get; set; }
    public decimal EstimatedCost { get; set; }
    public long InputTokens { get; set; }
    public long OutputTokens { get; set; }
    public long CachedInputTokens { get; set; }
    public List<AICostByDayModel> ByDay { get; set; } = [];
    public List<AICostByModelModel> ByModel { get; set; } = [];
    public List<AICostByPlanModel> ByPlan { get; set; } = [];
}

public class AICostByModelModel
{
    public string Model { get; set; } = string.Empty;
    public int RunCount { get; set; }
    public decimal EstimatedCost { get; set; }
}

public class AICostByPlanModel
{
    public string PlanCode { get; set; } = string.Empty;
    public int RunCount { get; set; }
    public decimal EstimatedCost { get; set; }
}
